universe/base.go

```diff
@@ -264,26 +268,70 @@ func (a *MintingArchive) RegisterNewIssuanceBatch(ctx context.Context,
 	log.Infof("Verifying %d new proofs for insertion into Universe",
 		len(items))
 
-	err := fn.ParSlice(
-		ctx, items, func(ctx context.Context, i *IssuanceItem) error {
-			assetSnapshot, err := a.verifyIssuanceProof(
-				ctx, i.ID, i.Key, i.Leaf,
-			)
-			if err != nil {
-				return err
-			}
+	// Issuances that also create an asset group, group anchors, must be
+	// verified and stored before any issuances that may be reissuances into
+	// the same asset group. This is required for proper verification of
+	// reissuances, which may be in this batch.
+	var anchorItems, nonAnchorItems []*IssuanceItem
```

Comment:
Same here re pre-allocating the slice that we expect the majority of items to be in.